### ISSUE TITLE
Symlink handling differs when scanning directories vs. specific files

### ISSUE THREAD

**Issue by malexmave** [semgrep/semgrep#4827]
**Describe the bug**
Semgreps behavior towards symlinked files differs between the directory-scanning and specific-file-scanning mode: A symlinked file can be scanned when scanning a directory it is contained in, but when targeting a scan directly on that file, semgrep claims it does not exist.

**To Reproduce**

Place the following in a file called test.py
```python
def danger2(request):
    # ruleid: command-injection-os-system
    image = request.POST['image']
    os.system("./face-recognize %s --N 24" % image)
```

Run the following command:
```bash
$ ln -s test.py test2.py
```

Scan the entire folder - both files are scanned:
```bash
$ semgrep -c p/ci .
Fetching rules from https://semgrep.dev/registry.
Scanning 2 files with 60 python rules.

Findings:

  test.py 
     python.django.security.injection.command.command-injection-os-system.command-injection-os-
     system
        Request data detected in os.system. This could be vulnerable to a command injection and
        should be avoided. If this must be done, use the 'subprocess' module instead and pass the
        arguments as a list. See https://owasp.org/www-community/attacks/Command_Injection for more
        information.
        Details: https://sg.run/Gen2

          6┆ image = request.POST['image']
          7┆ os.system("./face-recognize %s --N 24" % image)


  test2.py 
     python.django.security.injection.command.command-injection-os-system.command-injection-os-
     system
        Request data detected in os.system. This could be vulnerable to a command injection and
        should be avoided. If this must be done, use the 'subprocess' module instead and pass the
        arguments as a list. See https://owasp.org/www-community/attacks/Command_Injection for more
        information.
        Details: https://sg.run/Gen2

          6┆ image = request.POST['image']
          7┆ os.system("./face-recognize %s --N 24" % image)

Some files were skipped.
  Scan was limited to files tracked by git.

Ran 135 rules on 2 files: 2 findings.
```

Scan only the symlink - not found:
```bash
$ semgrep -c p/ci test2.py
Fetching rules from https://semgrep.dev/registry.
File not found: test2.py
```

**Expected behavior**
I expect the behavior to be consistent - either symlinks can be scanned, or they can't, but the behavior should not differ based on if I am giving the folder as a parameter, or the exact path to the symlink itself.

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [x] P1: important to fix or quite annoying (because it messes with an integration test)
- [ ] P2: regular bug that should get fixed

**Environment**
Mac OS, semgrep 0.85.0, installed via Homebrew.

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-1001/symlink-handling-differs-when-scanning-directories-vs-specific-files.

**Comment by underyx** [comment]
Thanks for the report @malexmave and sorry about the regression! Turns out our unit test for this behavior was buggy, so we didn't realize our refactor broke this behavior. The fix should go out next Tuesday with 0.86.
